backend.py

```diff
@@ -556,6 +597,47 @@ def _apply_cards_and_tiles_effects(self, registers):
 
             self.apply_tile_effects(register)
 
+    def create_card_pack(self):
+        """
+        Create and shuffle pack of cards: 42 movement and 42 rotation cards
+        with different values and priorities.
+        """
+        movement_cards = [(-1, 6, 250),
+                          (1, 18, 300),
+                          (2, 12, 400),
+                          (3, 6, 500),
+                          ]
+        rotation_cards = [(Rotation.U_TURN, 6, 50),
+                          (Rotation.LEFT, 18, 100),
+                          (Rotation.RIGHT, 18, 200),
+                          ]
+        self.new_card_pack = []
+
+        for movement, cards_count, first_number in movement_cards:
+            for i in range(cards_count):
+                # [MovementCard(690, -1)...][]
+                self.new_card_pack.append(MovementCard(first_number + i*5, movement))
+
+        for rotation, cards_count, first_number in rotation_cards:
+            for i in range(cards_count):
+                # [RotationCard(865, Rotation.LEFT)....]
+                self.new_card_pack.append(RotationCard(first_number + i*5, rotation))
+        shuffle(self.new_card_pack)
+        return self.new_card_pack
+
+    def get_dealt_cards(self, robot):
+        """
+        Deal the cards for robot - he gets one card less for every damage he's got.
+        Take and return the first cards from the card pack.
+        Delete the dealt cards from the card pack.
+        """
+        # Maximum number of cards is 9.
+        # Robot's damages reduce the count of dealt cards - each damage one card.
+        dealt_cards_count = MAX_CARD_COUNT-robot.damages
```

Comment:
@encukou  je to tady dobře, když se ten robot dává jako argument ne?

Reply:
Jo, tady je jasno jakého robota jde :)